src/python/WMComponent/DBS3Buffer/DBSBufferUtil.py

```diff
@@ -123,6 +123,19 @@ def loadBlocks(self, blocknames):
 
         return result
 
+    def loadDataset(self, datasetName):
+        """
+        Given a dataset name, load its information from DBSBuffer
+        :param datasetName: string with the dataset name
+        :return: a dictionary with the dataset info
+        """
+        loadDataset = self.daoFactory(classname="ListDataset")
+        result = loadDataset.execute(datasetPath=datasetName, transaction=False)
+        if not result:
+            # this should never happen, but just in case...
+            return {}
+        return result[0]
```

Comment:
Yes, changing the DAO would be another option. But I decided to make this change just in case this DAO is also used in other parts of the code, thus avoid extra changes. Still, there is a check for a non-empty list in the lines above it.